src/coreclr/System.Private.CoreLib/src/System/StubHelpers.cs

```diff
@@ -1219,9 +1219,16 @@ internal static Exception GetHRExceptionObject(int hr)
 #if FEATURE_COMINTEROP
         internal static Exception GetCOMHRExceptionObject(int hr, IntPtr pCPCMD, object pThis)
         {
-            Exception ex = InternalGetCOMHRExceptionObject(hr, pCPCMD, pThis);
-            ex.InternalPreserveStackTrace();
-            return ex;
+            if (Marshal.IsComSupported)
+            {
+                Exception ex = InternalGetCOMHRExceptionObject(hr, pCPCMD, pThis);
+                ex.InternalPreserveStackTrace();
+                return ex;
+            }
+            else
+            {
+                throw new NotSupportedException(SR.NotSupported_COM);
```

Comment:
This feels dangerous - I don't know when we call this `GetCOMHRExceptionObject`, but if it's called during exception processing, then we should not be throwing from it...